common/src/primitives/id/with_id.rs

```diff
@@ -29,6 +29,26 @@ pub struct WithId<T: Idable> {
     object: T,
 }
 
+impl<T> PartialEq for WithId<T>
+where
+    T: Idable,
+    T: PartialEq,
+    <T as Idable>::Tag: PartialEq,
+{
+    fn eq(&self, other: &WithId<T>) -> bool
+where {
+        self.id == other.id
+    }
+}
```

Comment:
I don't think it's a good idea to introduce the instance with this particular implementation. Ideally, the `WithId` type behaves completely transparently, i.e. `(WithId::new(x) == WIthId::new(y)) == (x == y)` for all `x` and `y`. In other words, the `==` operator should behave the same on `WithId` as on the original value. Sadly in `Transaction` some fields are excluded from ID, so the property is violated.

Conclusions:
* If you want an `Eq + PartialEq` instance on `WithId`, just let it generate the auto-derived version
* If you want to compare by IDs, use something like `x.get_id() == y.get_id()`, it's more explicit about what exactly is being compared.